typescript/token/contracts/extensions/HypNativeScaled.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.0;
+
+import {HypNative} from "../HypNative.sol";
+import {TokenRouter} from "../libs/TokenRouter.sol";
+
+/**
+ * @title Hyperlane Native Token that scales native value by a fixed factor for consistency with other tokens.
```

Comment:
Given how complicated it was for us to reason about the direction of the scale, I think a bit more documentation here would be appreciated